packages/go_router/lib/src/configuration.dart

```diff
@@ -347,14 +347,17 @@ class RouteConfiguration {
   /// Processes redirects by returning a new [RouteMatchList] representing the new
   /// location.
   FutureOr<RouteMatchList> redirect(
-      BuildContext context, FutureOr<RouteMatchList> prevMatchListFuture,
-      {required List<RouteMatchList> redirectHistory}) {
+    BuildContext context,
+    FutureOr<RouteMatchList> prevMatchListFuture, {
+    required List<RouteMatchList> redirectHistory,
+    bool forceRedirect = false,
+  }) {
     FutureOr<RouteMatchList> processRedirect(RouteMatchList prevMatchList) {
       final String prevLocation = prevMatchList.uri.toString();
       FutureOr<RouteMatchList> processTopLevelRedirect(
           String? topRedirectLocation) {
         if (topRedirectLocation != null &&
-            topRedirectLocation != prevLocation) {
+            (forceRedirect || topRedirectLocation != prevLocation)) {
```

Comment:
I am not entirely sure why we need this flag. if the location is the same wouldn't the newMatch be the same after `_getNewMatches`